flags/src/main/java/com/yahoo/vespa/flags/JacksonFlag.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.flags;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.function.Function;
+
+import static com.yahoo.yolean.Exceptions.uncheck;
+
+/**
+ * @author hakonhall
+ */
+public class JacksonFlag<T> implements Flag {
+    private final static ObjectMapper mapper = new ObjectMapper();
+
+    private final FlagId id;
+    private final Class<T> jacksonClass;
+    private final T defaultValue;
+    private final FlagSource source;
+
+    public static <T> Function<FlagSource, JacksonFlag<T>> createUnbound(String flagId, Class<T> jacksonClass, T defaultValue) {
+        return createUnbound(new FlagId(flagId), jacksonClass, defaultValue);
+    }
+
+    public static <T> Function<FlagSource, JacksonFlag<T>> createUnbound(FlagId id, Class<T> jacksonClass, T defaultValue) {
+        return source -> new JacksonFlag<>(id, jacksonClass, defaultValue, source);
+    }
+
+    public JacksonFlag(String flagId, Class<T> jacksonClass, T defaultValue, FlagSource source) {
```

Comment:
Can't we just get the class from the `defaultValue`?

Reply:
I don't think so, e.g. defaultValue.getClass() returns the type Class<? extends Object>.

Reply:
Can't it safely be casted to `Class<T>`?

Reply:
The best I could come up with is:
https://stackoverflow.com/questions/3437897/how-to-get-a-class-instance-of-generics-type-t

which recommends this type of API.

Reply:
https://stackoverflow.com/questions/36390184/how-to-get-class-instances-of-generic-types-in-generic-methods

recommends adding Class<T> parameter to a method taking 'T instance', in order to get an Class<T> instance.

Reply:
Right, but in this case we do take in a default value which is an instance of `T`, so it should be safe to this:
```
 public JacksonFlag(String flagId, T defaultValue, FlagSource source) {
    Class<T> jacksonClass = (Class<T>) defaultValue.getClass();
    ...
```
?